lua/rzls/documentstore.lua

```diff
@@ -84,6 +84,21 @@ function M.register_vbufs(source_buf)
     return M.register_vbufs_by_path(currentFile)
 end
 
+---Refreshes parent views of the given virtual document
+---@param result VBufUpdate
+function M.refresh_parent_views(result)
+    local uri = vim.uri_from_fname(result.hostDocumentFilePath)
+    ---@type rzls.VirtualDocument?
+    local rvd = virtual_documents[uri]
+    if not rvd or rvd.kind ~= razor.language_kinds.razor then
+        assert(false, "Not a razor document")
+        return
+    end
+    if vim.lsp.inlay_hint.is_enabled({ bufnr = rvd.buf }) then
```

Comment:
This change fixes the desync issues I was seeing. I was a little confused why this works but I see, `vim.lsp.inlay_hint.enable` also calls `_refresh` to refresh inlay hints when it's called. I think the lsp module doesn't expose a dedicated `inlay_hint.refresh` method because it already hooks up autocommands to refresh inlay hints on LspNotify textDocumentChange. I'll create an issue to look into why that "normal" flow isn't working because it suggests something is not quite right somewhere...